cmd/s3-layer.go

```diff
@@ -77,73 +78,68 @@ func s3ToObjectError(err error, params ...string) error {
 		}
 	case "XMinioInvalidObjectName":
 		err = ObjectNameInvalid{}
+	case "AccessDenied":
+		err = PrefixAccessDenied{
+			Bucket: bucket,
+			Object: object,
+		}
 	default:
-		fmt.Println("Undefined s3 object error:", bucket, object, minioErr.Code)
+		// should not be reached
+		err = fmt.Errorf("Undefined s3 object error: bucket=%s object=%s code=%s", bucket, object, minioErr.Code)
```

Comment:
This will cause the original error to be replaced with fmt.Errorf() error, hence you could just remove the entire `default` section from the switch statement. There is a high chance of us reaching default statement as number `if` cases in the switch statement is limited.